syntax = "proto3";
package user_proto;
import "google/protobuf/empty.proto";

service UserService {
  rpc GetUser (UserRetrieveRequest) returns (User) {};
  rpc ListUsers (UserListRequest) returns (stream User) {};
  rpc CreateUser (User) returns (User) {};
  rpc UpdateUser (User) returns (User) {};
  rpc DeleteUser (User) returns (google.protobuf.Empty) {};
}

message User {
  string id = 1;
  string email = 2;
  string firstname = 3;
  string lastname = 4;
  string role = 7;
  bool is_active = 8;
}

message UserRetrieveRequest {
  string id = 1;
}

message UserListRequest {}


